Makefile

```diff
@@ -0,0 +1,90 @@
+SHELL=/bin/bash
+DATETIME:=$(shell date -u +%Y%m%dT%H%M%SZ)
+
+.PHONY: help install update test coveralls
+
+# Discover Javascript/Node projects: immediate subdirectories that contain 
+# a nodejs/package.json
+PROJECT_DIRS := $(shell for d in ./*/ ; do if [ -f "$$d/nodejs/package.json" ]; then echo $${d%/} ; fi ; done)
+
+help: # Preview Makefile commands
+	@awk 'BEGIN { FS = ":.*#"; print "Usage:  make <target>\n\nTargets:" } \
+		/^[-_[:alpha:]]+:.?*#/ { printf "  %-15s%s\n", $$1, $$2 }' $(MAKEFILE_LIST)
+
+
+#######################
+# Dependency commands
+#######################
+
+install: # Install Node.js dependencies for each project (uses npm ci)
+	@echo "Projects discovered: $(PROJECT_DIRS)"
+	@if [ -z "$(PROJECT_DIRS)" ]; then \
+		echo "No projects found to install dependencies for."; exit 1; \
+	fi
+	@for p in $(PROJECT_DIRS); do \
+		echo "\n==> Installing dependencies for $$p"; \
+		cd $$p/nodejs && \
+		if [ -f package-lock.json ]; then \
+			npm ci --no-audit --no-fund; \
+		else \
+			npm install --no-audit --no-fund; \
+		fi || exit $$?; \
+	done
+
+
+update: # Update Node.js dependencies for each project (npm update + prune)
+	@echo "Projects discovered: $(PROJECT_DIRS)"
+	@if [ -z "$(PROJECT_DIRS)" ]; then \
+		echo "No projects found to update dependencies for."; exit 1; \
+	fi
+	@for p in $(PROJECT_DIRS); do \
+		echo "\n==> Updating dependencies for $$p"; \
+		cd $$p/nodejs && \
+		npm update --no-audit --no-fund && \
+		npm prune --no-audit --no-fund || { echo "Update failed for $$p"; exit 1; }; \
+	done
+
+
+######################
+# Unit test commands
+######################
+
+test: # Run tests (with coverage) for each discovered project. Use PROJ or PROJS to limit.
+	@echo "Projects discovered: $(PROJECT_DIRS)"
+	@if [ -z "$(PROJECT_DIRS)" ]; then \
+		echo "No projects found to test (looked for nodejs/package.json in subfolders)."; exit 1; \
+	fi
+	@for p in $(PROJECT_DIRS); do \
+		echo "\n==> Running tests for $$p"; \
+		cd $$p/nodejs && \
+		if [ ! -d node_modules ]; then \
+			echo "Run 'make install' first!"; \
+		fi && \
+		NODE_ENV=test npm test -- --coverage --coverageReporters=lcov --coverageReporters=text || exit $$?; \
+	done
+
+
+######################
+# Local "deploy" commands
+######################
+
+# Use a cli parameter to denote the folder (project) where the code resides.
+# Additionally, expect that the local ENV has S3BUCKET set so this command knows
+# where to copy the zip file.
+PROJ ?= 
+
+deploy-dev: # Zip and copy to S3 bucket
```

Comment:
Minor: I might break this into multiple make commands. Being able to generate the zip file(s) separately from deploying them may be useful for debugging. For example, having the zip creation be `make dist` and then `make deploy-dev` could call `dist` to ensure the files are created before deploying them.

Possibly even a `make clean` to handle the clean up for the zip files as it's own command would be good practice.